src/sbvr-api/permissions.ts

```diff
@@ -255,21 +255,29 @@ export function nestedCheck<I, O>(
 	throw new Error('Cannot parse required checks: ' + check);
 }
 
+interface CollapsedFilter<T> extends Array<string | T | CollapsedFilter<T>> {
+	0: string;
+}
+
 const collapsePermissionFilters = <T>(
 	v: NestedCheck<{ filter: T }>,
-): T | [string, T] => {
+): T | CollapsedFilter<T> => {
 	if (_.isArray(v)) {
 		return collapsePermissionFilters({ or: v });
 	}
 	if (typeof v === 'object') {
 		if ('filter' in v) {
 			return v.filter;
 		}
-		return _(v)
-			.toPairs()
-			.flattenDeep()
-			.map(collapsePermissionFilters)
-			.value() as [string, T];
+		if ('and' in v) {
+			return ['and', ...v.and.map(collapsePermissionFilters)];
+		}
+		if ('or' in v) {
+			return ['or', ...v.or.map(collapsePermissionFilters)];
+		}
+		throw new Error(
+			'Permission filter objects must have `filter` or `and` or `or` keys',
```

Comment:
Does this change still count as `Change-type: patch`? (Not saying it doesn't, just a thought...)

Reply:
It should do, it should be impossible to ever hit this branch because it's based on the output of `nestedCheck` so right now this is unreachable code, but it's here for typescript to verify the types properly and also to make sure that other changes within pinejs don't break things in non-obvious ways (ie they'll hit this error instead of weirdness around `undefined` where it's unexpected)